### DISCUSSION TITLE
What kind of database/filetype is the db.zo file?

### DISCUSSION THREAD

**Discussion by kartikdc** [ajeetdsouza/zoxide#366] [Q&A] [answered]
I have never seen a file like that so it got me wondering. Might be useful for me in the future. Thanks.

**Comment by ajeetdsouza** [comment, marked as answer]
It's a binary format. There's a 4-byte header that gives you the database version (currently version 3) followed by the serialized list of directories (path, rank, and timestamp). The code for it is [here](https://github.com/ajeetdsouza/zoxide/blob/17365710af7cfb9cfe0030756ac31ffae4286436/src/db/dir.rs#L10), and the format used for serialization is called [bincode](https://github.com/bincode-org/bincode).

**Comment by bfrontend** [comment]
Hello, I would like to know how to deserialize this data file in another language such as python or nodejs

**Reply by ajeetdsouza** [reply]
Could you describe your use case?

**Reply by bfrontend** [reply]
I wrote a extension for vscode that calls a different smart command data file to quickly open the folder in vscode and update the data file. Currently support the z/autojump, this is the address of the repository this extension https://github.com/webxmsj/vscode-autojump

**Reply by ajeetdsouza** [reply]
Have you tried the CLI? Check out `zoxide query --help` and `zoxide add --help`.

A lot of third party plugins have been written that use zoxide (listed on the README), they all leverage the CLI.

**Reply by bfrontend** [reply]
Ok, thanks. I'll go see the cli way
